src/evm/stream.rs

```diff
@@ -52,33 +68,42 @@ impl ProtocolStreamBuilder {
         self
     }
 
+    /// Sets the block time for the Tycho client.
     pub fn block_time(mut self, block_time: u64) -> Self {
         self.stream_builder = self
             .stream_builder
             .block_time(block_time);
         self
     }
 
+    /// Sets the timeout duration for network operations.
     pub fn timeout(mut self, timeout: u64) -> Self {
         self.stream_builder = self.stream_builder.timeout(timeout);
         self
     }
 
+    /// Configures the client to exclude state updates from the stream.
     pub fn no_state(mut self, no_state: bool) -> Self {
         self.stream_builder = self.stream_builder.no_state(no_state);
         self
     }
 
+    /// Sets the API key for authenticating with the Tycho server.
     pub fn auth_key(mut self, auth_key: Option<String>) -> Self {
         self.stream_builder = self.stream_builder.auth_key(auth_key);
         self
     }
 
+    /// Disables TLS/ SSL for the connection, using http and ws protocols.
     pub fn no_tls(mut self, no_tls: bool) -> Self {
         self.stream_builder = self.stream_builder.no_tls(no_tls);
         self
     }
 
+    /// Sets the currently known tokens which to be considered during decoding.
+    ///
+    /// Protocol components containing tokens which are not included in this initial list, or
+    /// added when applying deltas, will not be decoded.
```

Comment:
This is so crucial information it should really be mentioned on the structs docs.